interview-questions.md

```diff
@@ -8,33 +8,33 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Your answer:
+Your answer: The main difference between a parameter and an argument is that the parameter is  placeholder for data that is passed into a function, while an argument is the actual data that is passed through into the function.
 
-Researched answer:
+Researched answer: When referring to JavaScript functions, parameters and arguments are used interchangeably, however this is not the case, and there is a very suble difference. Parameters are variables listed within the function definition, whereas arguements are values that are PASSED to the function when invoked.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: Predefined parameters in the .map() method are value, index, and array, and I think that they are all required.
 
-Researched answer:
+Researched answer: The method .map() takes the predefined parameters of function (req'd), currentValue (req'd), arr, index, and thisValue with the last three listed being the optional.
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: The main difference between map and filter methods is that although both returns a new array from the original subset, with map, it returns back every value from original array, with the new conditon we assigned it, whereas with filter, it returns a new array as the name implies, filtering and returning only what we want back.
 
-Researched answer:
+Researched answer: Upon researching the key differences between map and filter, multiple resources states what I described above, but using different words and a little additional detail. For example, with map, it will always return the same number elemenets as the original, however with the element values transformed in some way, while filter can sometimes return the same, or less amount of elements than the original, and will not change the original element values.
 
 4. What is iteration?
 
-Your answer:
+Your answer: Iteration is the action of repeating a function over and over until a specified action has been met wheter it be a string, or an array.
 
-Researched answer:
+Researched answer: Iteration in JavaScript refers to a given sequence of instructions and code being repeated until a specific end result is achived.
 
 5. What is the difference between a class and an object?
 
-Your answer:
+Your answer: Classes can be considered as a "recipe" for making the actual objects, whereas objects are a JavaScript data type containing keys and values.
 
-Researched answer:
+Researched answer: Classes are detailed, descriptive, and used as a template for what an object will be, however itself, is not an object. Essentially a blueprint of what the object is. Objects are a collection of different properties held within a key and a value. Where they differ is that classes are grouped objects that are similiar
 
 6. STRETCH: What is hoisting in JavaScript?
 
```

Comment:
Take an opportunity to check out what hoisting is. It might be beneficial in future projects.